File: Hospital_Management_Microservice/Hospital_Management_NET/Hospital_Management/Models/Appointment.cs
using System;
using System.Collections.Generic;

namespace Hospital_Management.Models;

public partial class Appointment
{
    public long Id { get; set; }

    public DateTime? AppointmentDate { get; set; }

    public string? DiseaseDescription { get; set; }

    public string? Status { get; set; }

    public long? DoctorId { get; set; }

    public long? PatientId { get; set; }

    public virtual Doctor? Doctor { get; set; }

    public virtual Patient? Patient { get; set; }

    public virtual ICollection<Payment> Payments { get; set; } = new List<Payment>();
}

File: Hospital_Management_Microservice/hospital management_springboot/src/main/java/com/app/Entity/Appointment.java
package com.app.Entity;

import java.time.LocalDate;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Appointment {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne
	@JoinColumn(name = "patient_id")
	@JsonIgnore
	private Patient patient;

	@ManyToOne
	@JoinColumn(name = "doctor_id")
	@JsonIgnore
	private Doctor doctor;

	private LocalDate appointmentDate;
	private String diseaseDescription;
	private String status; // PENDING, ACCEPTED, REJECTED

	@OneToOne(mappedBy = "appointment", cascade = CascadeType.ALL)
	private Payment payment;

}
